receiver/cloudfoundryreceiver/receiver.go

```diff
@@ -93,13 +133,22 @@ func (cfr *cloudFoundryReceiver) Start(ctx context.Context, host component.Host)
 			return
 		}
 
-		envelopeStream, err := streamFactory.CreateStream(innerCtx, cfr.config.RLPGateway.ShardID)
+		envelopeStream, err := streamFactory.CreateStream(innerCtx, cfr.config.RLPGateway.ShardID, cfr.telemetryType)
 		if err != nil {
 			cfr.settings.ReportStatus(component.NewFatalErrorEvent(fmt.Errorf("creating RLP gateway envelope stream: %w", err)))
 			return
 		}
 
-		cfr.streamMetrics(innerCtx, envelopeStream)
+		switch cfr.telemetryType {
```

Comment:
One thing we could actually do, we could remove `telemetryType`, and then do nil checks for `cfr.nextMetricsConsumer` and `cfr.nextLogsConsumer` instead. This may be more simple than an extra variable.

The otlp receiver does this, as [a reference](https://github.com/open-telemetry/opentelemetry-collector/blob/e55d22aec61d7b9787b88448ec4a2ae8fb5eefd6/receiver/otlpreceiver/otlp.go#L126).